src/main/java/uk/gov/pay/ledger/agreement/dao/AgreementDao.java

```diff
@@ -87,6 +89,22 @@ public class AgreementDao {
             "FROM agreement a " +
             ":searchExtraFields";
 
+    private static final String SELECT_ASSOCIATED_EVENTS = "SELECT results.* FROM " +
+            "(" +
+                "SELECT e.id, e.sqs_message_id, e.service_id, e.live, 'payment_instrument' AS resource_type_name, " +
+                "e.resource_external_id, e.parent_resource_external_id, e.event_date, e.event_type, e.event_data " +
+                "FROM agreement a " +
```

Comment:
We don't need to join agreement table as no data is needed from it. I think querying just by agreement ID should be fine

Join between payment_instrument and event table is an inner join. Worth checking if index on event table is being used for this query

Reply:
Yes, there is an index on event.resource_external_id.

No data is selected from the agreement table, but it is needed in the join because it is used for filtering by serviceId.  I could get rid of it if it is the case that the serviceId column in the event table will always be populated for payment instrument events.  I don't *think* this is the case, but I may be wrong - do you know?

Reply:
On reflection, I can get rid of the join if I just stop checking the serviceId in the query entirely - it's already checked when we look up the agreement entity in the service (to ensure the agreement exists).